core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -335,16 +348,23 @@ impl SchedulerController {
             let post_lock_validation_count = transactions.len();
 
             let mut post_transaction_check_count: usize = 0;
-            for ((transaction, priority_details), _) in transactions
+            for ((transaction, budget_limits), _) in transactions
                 .into_iter()
-                .zip(priority_details)
+                .zip(budget_limits)
                 .zip(check_results)
                 .filter(|(_, check_result)| check_result.0.is_ok())
             {
                 saturating_add_assign!(post_transaction_check_count, 1);
                 let transaction_id = self.transaction_id_generator.next();
 
-                let transaction_cost = CostModel::calculate_cost(&transaction, &bank.feature_set);
+                let cost = CostModel::calculate_cost(&transaction, &bank.feature_set).sum();
+                let priority = Self::calculate_prioritization(
+                    fee_structure,
+                    include_loaded_account_data_size_in_fee,
+                    transaction.message(),
+                    &budget_limits,
+                    cost,
+                );
```

Comment:
imo, this call-site should _look_ like this to illustrate its high level relationships for humans at a quick glance:

```rust
let (priority, cost) = Self::calculate_priority_and_cost(
    &transaction,
    fee_structure,
    &compute_budget_limits,
    // include_loaded_account_data_size_in_fee // <= depends on perf. penalty of .feature_set().is_active(...) invoked everytime for txes.
)
```

then, the call-hierarchy should like like this
- calculate_priority_and_cost()
  - calculate_fee()
  - calculate_cost()
  - `fee.saturating_mul(MULTIPLIER).saturating_div(cost)`

these 2 new calls are complex in shape, i think hiding details in this context (fn named as _`buffer_packets()`_) should improve readability.